src/plugins/explore/public/application/utils/state_management/actions/set_dataset/set_dataset.ts

```diff
@@ -40,6 +42,8 @@ export const setDatasetActionCreator = (
   dispatch(clearResults());
   dispatch(clearQueryStatusMap());
   dispatch(clearLastExecutedData());
+  dispatch(setPatternsField(''));
```

Comment:
i would just use https://github.com/kavilla/OpenSearch-Dashboards-1/blob/kavilla/dsselect_move/src/plugins/explore/public/application/utils/state_management/middleware/dataset_change_middleware.ts#L55

Reply:
using this